examples/documentation_examples/examples.go

```diff
@@ -2956,3 +2957,172 @@ func StableAPIExamples() {
 	StableAPINonStrictExample()
 	StableAPIDeprecationErrorsExample()
 }
+
+func insertSnapshotQueryTestData(mt *mtest.T) {
+	catColl := mt.CreateCollection(mtest.Collection{Name: "cats"}, true)
+	_, err := catColl.InsertMany(context.Background(), []interface{}{
+		bson.D{
+			{"adoptable", false},
+			{"name", "Miyagi"},
+			{"color", "grey-white"},
+			{"age", 14},
+		},
+		bson.D{
+			{"adoptable", true},
+			{"name", "Joyce"},
+			{"color", "black"},
+			{"age", 10},
+		},
+	})
+	require.NoError(mt.T, err)
+
+	dogColl := mt.CreateCollection(mtest.Collection{Name: "dogs"}, true)
+	_, err = dogColl.InsertMany(context.Background(), []interface{}{
+		bson.D{
+			{"adoptable", true},
+			{"name", "Cormac"},
+			{"color", "rust"},
+			{"age", 7},
+		},
+		bson.D{
+			{"adoptable", true},
+			{"name", "Frank"},
+			{"color", "yellow"},
+			{"age", 2},
+		},
+	})
+	require.NoError(mt.T, err)
+
+	salesColl := mt.CreateCollection(mtest.Collection{Name: "sales"}, true)
+	_, err = salesColl.InsertMany(context.Background(), []interface{}{
+		bson.D{
+			{"shoeType", "hiking boot"},
+			{"price", 30.0},
+			{"saleDate", time.Now()},
+		},
+	})
+	require.NoError(mt.T, err)
+}
+
+func snapshotQueryPetExample(mt *mtest.T) error {
+	client := mt.Client
+	db := mt.DB
+
+	// Start Snapshot Query Example 1
+	ctx := context.TODO()
+
+	sess, err := client.StartSession(new(options.SessionOptions).SetSnapshot(true))
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var adoptablePetsCount int32
+	err = mongo.WithSession(ctx, sess, func(ctx mongo.SessionContext) error {
+		// Count the adoptable cats
+		adoptableCatsOutput := "adoptableCatsCount"
+		cursor, err := db.Collection("cats").Aggregate(ctx, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableCatsOutput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(ctx)
+		resp := cursor.Current.Lookup(adoptableCatsOutput)
+		adoptableCatsCount, ok := resp.Int32OK()
+		if !ok {
+			return fmt.Errorf("failed to find int32 field %q in document %v", adoptableCatsOutput, cursor.Current)
+		}
+		adoptablePetsCount += adoptableCatsCount
+
+		// Count the adoptable dogs
+		adoptableDogsOutput := "adoptableDogsCount"
+		cursor, err = db.Collection("dogs").Aggregate(ctx, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableDogsOutput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(ctx)
+		resp = cursor.Current.Lookup(adoptableDogsOutput)
+		adoptableDogsCount, ok := resp.Int32OK()
+		if !ok {
+			return fmt.Errorf("failed to find int32 field %q in document %v", adoptableDogsOutput, cursor.Current)
+		}
+		adoptablePetsCount += adoptableDogsCount
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	// End Snapshot Query Example 1
+	require.Equal(mt.T, int32(3), adoptablePetsCount, "expected adoptablePetsCount to be 3, got %v", adoptablePetsCount)
```

Comment:
Optional: `require.Equal` prints rich assertion failures, including the expected and actual values, so you can simplify this line.

E.g.
```go
require.Equal(mt, int32(3), adoptablePetsCount, "expected 3 total adoptable pets")
```